### ISSUE TITLE
In some cases, mods cannot be downloaded by using mod metadata.

### ISSUE THREAD

**Issue by cneicy** [PrismLauncher/PrismLauncher#3218]
### Operating System

Windows

### Version of Prism Launcher

Prism Launcher 9.1

### Version of Qt

Qt 6.7.3

### Description of bug

If the mod's file on CurseForge does not have a modloader set, 
![image](https://github.com/user-attachments/assets/706fabf2-a8cb-46cc-8a4e-f2bd60b7818b)
![image](https://github.com/user-attachments/assets/9958cf24-367f-4a6e-add5-ee23d0e2880f)
the data in the mod's metadata under `x-prismlauncher-loaders` will be empty. 
![image](https://github.com/user-attachments/assets/abdbe6bc-31c8-4261-9f49-fe215a13a87c)
![image](https://github.com/user-attachments/assets/87cb0803-daaa-4530-8e95-25bf2195dcd4)
As a result, when using this mod metadata to download the mod(select it and click `Check for Updates` ), it cannot be downloaded, and you may encounter the error: "No valid version found for this mod. It's probably unavailable for the current game version / mod loader."
![image](https://github.com/user-attachments/assets/f1e6c889-7c93-4ca5-b3d0-5f6410389f9e)
and...
![image](https://github.com/user-attachments/assets/8bfb6f21-f2ff-4708-9a76-edcfc49bd375)

Here is the expected one 
![image](https://github.com/user-attachments/assets/dadd9296-d023-47be-9488-dc888d7fce25)

### Steps to reproduce

[.index.zip](https://github.com/user-attachments/files/18183211/default.index.zip)
[screen record](https://forum.crash.work/assets/files/2024-10-25/prism.mp4)

### Suspected cause

It might be caused by an update to packwiz.
This issue only occurs in version 9.1, while users on version 8.X do not encounter this problem.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Trial97** [comment]
Can you reproduce this with the build found here: https://github.com/PrismLauncher/PrismLauncher/actions/runs/11629379869?pr=3043 (from this PR: https://github.com/PrismLauncher/PrismLauncher/pull/3043)
As this sounds exactly the same with https://github.com/PrismLauncher/PrismLauncher/issues/3040

**Comment by cneicy** [comment]
> Can you reproduce this with the build found here: https://github.com/PrismLauncher/PrismLauncher/actions/runs/11629379869?pr=3043 (from this PR: #3043) As this sounds exactly the same with #3040

it works. thank you

**Comment by Trial97** [comment]
Ok then I will close this as duplicate of the mentioned issue

**Comment by cneicy** [comment]
> Can you reproduce this with the build found here: https://github.com/PrismLauncher/PrismLauncher/actions/runs/11629379869?pr=3043 (from this PR: #3043) As this sounds exactly the same with #3040

but for mods like `Scape and Run: Parasites`, which do not allow third-party downloads, it will directly result in an error.

![image](https://github.com/user-attachments/assets/964e7f87-bf53-460a-b099-32524bba8653)
when encountering mods like `Scape and Run: Parasites` that don't allow third-party downloads, then pop up the box directing the user to CurseForge is better.

**Comment by Trial97** [comment]
The last part sounds more like a feature than a bug report: 
Is the error message happening for blocked mods with a loader (in the meta),or only for the ones that do not have the loader information available?
It would be better to make a feature request for the blocked mods dialog. 

I will try to look into the fact that you get the error message with an empty URL there and address it in a minor version.
And keep the blocked mods dialog feature for a major release.

**Comment by cneicy** [comment]
> The last part sounds more like a feature than a bug report: Is the error message happening for blocked mods with a loader (in the meta),or only for the ones that do not have the loader information available? It would be better to make a feature request for the blocked mods dialog.
> 
> I will try to look into the fact that you get the error message with an empty URL there and address it in a minor version. And keep the blocked mods dialog feature for a major release.

Yes, that part is a feature request. Mods that do not allow third-party downloads on CurseForge will throw an error when trying to download via mod metadata, as shown in the screenshot above. My suggestion is for PrismLauncher to work like when importing CurseForgePack, where it provides a download link for mods that block third-party downloads, allowing users to download them directly from CurseForge.

Regardless of whether there is a loader in the x-prismlauncher-loaders metadata, any mod that blocks third-party downloads will encounter this problem.

**Comment by Trial97** [comment]
@cneicy  double checking are you getting that Task failed dialog on the nightly build: https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop
I'm trying to reproduce it but I constantly get a dialog mentioning that the mod is up to date(or another one that there is an update and I should update:
![image](https://github.com/user-attachments/assets/7745ce93-d8e3-49ba-bc7f-1ab33e7c0b0f)
PS is that dialog ^^ enough for blocked mods issue we discussed?

**Comment by cneicy** [comment]
> @cneicy double checking are you getting that Task failed dialog on the nightly build: https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop I'm trying to reproduce it but I constantly get a dialog mentioning that the mod is up to date(or another one that there is an update and I should update: ![image](https://private-user-images.githubusercontent.com/42720688/397126920-7745ce93-d8e3-49ba-bc7f-1ab33e7c0b0f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MDU1NjQsIm5iZiI6MTczNDYwNTI2NCwicGF0aCI6Ii80MjcyMDY4OC8zOTcxMjY5MjAtNzc0NWNlOTMtZDhlMy00OWJhLWJjN2YtMWFiMzNlN2MwYjBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDEwNDc0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5OGI4YzM4MWRkMTgxZDE2MDZmYjYyZWZhYzQzNDU4ZGVjMTIwZjRmYTkxOTZlMDI3YTlkZTFiNzJhYmU5MTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.u26sgxqCvlZjx3rvAcsD0xG3K1IiakVnO6Q-K-N9JRo) PS is that dialog ^^ enough for blocked mods issue we discussed?

https://github.com/user-attachments/assets/68c730d7-13ec-4914-a66d-a00d484723f2
